onnxruntime/python/tools/quantization/calibrate.py

```diff
@@ -134,7 +134,7 @@ def get_intermediate_outputs(self, calib_mode='naive', providers=None):
         '''
 
         #conduct inference session and get intermediate outputs
-        if providers:
+        if tensorrt_calibration:
```

Comment:
If we add this tensorrt_calibration variable, should we also modify "calibrator.get_intermediate_outputs(providers=["CUDAExecutionProvider"]) " in calculate_calibration_data() as well?

Reply:
i'm not in favor of this approach. we should make calibrate as agnostic as possible. in theory it should be able to support the needs of different callers.
i.e.  the caller can already specify the providers 
and calibrate() can be modified to calculate quantization params or output the raw data.  (or later support multiple ways of calculating params)